### PR TITLE
Split clickhouse admin server

### LINKED ISSUE #6779: Split `clickhouse-admin` into 2 separate servers
Right now, the same `clickhouse-admin` service run in the `ClickhouseKeeper` and `ClickhouseServer` zones. However, each of those zones uses a disjoint set of endpoints which implies that we should have two separate admin servers: `clickhouse-admin-server` and `clickhouse-admin-keeper`.

### LINKED ISSUE #6829: remove omicron-nexus's transitive dependency on clickhouse-admin-api
The `ls-apis` tool assumes that if a deployed component (like `omicron-nexus`) depends on a server crate (like `clickhouse-admin-api`), then that component *exports* that API.  I think this is a reasonable invariant to try to maintain.  Right now though `omicron-nexus` does depend on `clickhouse-admin-api`, actually through two paths:

```
$ cargo tree -p omicron-nexus -i clickhouse-admin-api
clickhouse-admin-api v0.1.0 (/home/dap/omicron-work/clickhouse-admin/api)
├── clickhouse-admin-client v0.1.0 (/home/dap/omicron-work/clients/clickhouse-admin-client)
│   └── nexus-reconfigurator-execution v0.1.0 (/home/dap/omicron-work/nexus/reconfigurator/execution)
│       └── omicron-nexus v0.1.0 (/home/dap/omicron-work/nexus)
└── nexus-reconfigurator-execution v0.1.0 (/home/dap/omicron-work/nexus/reconfigurator/execution) (*)
```

even though it doesn't actually expose the `clickhouse-admin-api`.

I found this in trying to fix #6827 because if I just add metadata for `clickhouse-admin-client`, then `ls-apis` fails with:

```
error: API for client clickhouse-admin-client appears to be exported by multiple components: at least omicron-clickhouse-admin and omicron-nexus (DepPath([PackageId { repr: "path+file:///home/dap/omicron-work/nexus/reconfigurator/execution#nexus-reconfigurator-execution@0.1.0" }, PackageId { repr: "path+file:///home/dap/omicron-work/nexus#omicron-nexus@0.1.0" }]))
```

`ls-apis`'s data model does not currently allow this (and, in its defense, we don't _actually_ run the same API from different components today -- it's just an artifact of the way it's trying to figure out what servers are in which components).

### PR DESCRIPTION
**Pull request by andrewjstone**
Fixes #6779
Fixes #6829

### CODE REVIEW

**Comment by andrewjstone** [comment]
I'm pretty sure I have to fix some omicron-pkg and smf stuff here as well. I'll do that shortly.

File: clickhouse-admin/api/src/lib.rs [hunk 1]
```diff
@@ -95,3 +68,19 @@ pub trait ClickhouseAdminApi {
         rqctx: RequestContext<Self::Context>,
     ) -> Result<HttpResponseOk<ClickhouseKeeperClusterMembership>, HttpError>;
 }
+
+#[dropshot::api_description]
```
Comment (sunshowers) [inline clickhouse-admin/api/src/lib.rs:72]: Could you add a little comment here explaining why the two are different? Thinking of my future self which would be wondering why.

File: clickhouse-admin/types/src/lib.rs [hunk 2]
```diff
@@ -16,8 +21,20 @@ use std::io::{ErrorKind, Write};
...
+mod config;
+pub use config::{
+    ClickhouseHost, KeeperConfig, KeeperConfigsForReplica, KeeperNodeConfig,
+    LogConfig, LogLevel, Macros, NodeType, RaftServerConfig,
+    RaftServerSettings, RaftServers, ReplicaConfig, ServerNodeConfig,
+};
+
+// Used for schemars to be able to be used with camino:
+// See https://github.com/camino-rs/camino/issues/91#issuecomment-2027908513
+pub fn path_schema(gen: &mut SchemaGenerator) -> Schema {
```
Comment (sunshowers) [inline clickhouse-admin/types/src/lib.rs:33]: 😭

File: clients/clickhouse-admin-keeper-client/src/lib.rs [hunk 3]
```diff
@@ -2,11 +2,11 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-//! Interface for making API requests to a clickhouse-admin server running in
+//! Interface for making API requests to a clickhouse-admin-{keeper|server} server running in
```
Comment (sunshowers) [inline clients/clickhouse-admin-keeper-client/src/lib.rs:5]: Just `clickhouse-admin-keeper` here right?

File: dev-tools/ls-apis/src/system_apis.rs [hunk 4]
```diff
@@ -444,22 +444,6 @@ impl<'a> ServerComponentsTracker<'a> {
...
         }
 
-        // TODO Work around omicron#6829.
-        // Through the dependency tree, omicron-nexus appears to export the
-        // clickhouse-admin-api, but it doesn't really.
-        // This is just like the Crucible Pantry one above in that we can't
-        // build our data model unless we ignore it so we have to hardcode this
-        // here rather than use "dependency_filter_rules".
-        if **server_pkgname == "omicron-nexus"
-            && *api.client_package_name == "clickhouse-admin-client"
```
Comment (sunshowers) [inline dev-tools/ls-apis/src/system_apis.rs:454]: yay

File: dev-tools/openapi-manager/src/spec.rs [hunk 5]
```diff
@@ -27,14 +27,25 @@ pub fn all_apis() -> Vec<ApiSpec> {
...
             boundary: ApiBoundary::Internal,
             api_description:
-                clickhouse_admin_api::clickhouse_admin_api_mod::stub_api_description,
-            filename: "clickhouse-admin.json",
+                clickhouse_admin_api::clickhouse_admin_keeper_api_mod::stub_api_description,
+            filename: "clickhouse-admin-keeper.json",
+            extra_validation: None,
+        },
+        ApiSpec {
+            title: "ClickHouse Cluster Admin Keeper API",
```
Comment (sunshowers) [inline dev-tools/openapi-manager/src/spec.rs:41]:
```suggestion
            title: "ClickHouse Cluster Admin Server API",
```

**Review by sunshowers — COMMENTED** [review]
A few comments

File: clickhouse-admin/types/src/lib.rs (hunk 2 above)
Comment (andrewjstone) [reply clickhouse-admin/types/src/lib.rs:33]: LOL. It's everyone's* favorite issue reference!

*Rain's favorite

File: dev-tools/openapi-manager/src/spec.rs (hunk 5 above)
Comment (andrewjstone) [reply dev-tools/openapi-manager/src/spec.rs]: Fixed in [9862199](https://github.com/oxidecomputer/omicron/pull/6837/commits/98621999699a16737ee73df59b5f17c880dfcb7c)

File: clients/clickhouse-admin-keeper-client/src/lib.rs (hunk 3 above)
Comment (andrewjstone) [reply clients/clickhouse-admin-keeper-client/src/lib.rs]: Fixed in [9862199](https://github.com/oxidecomputer/omicron/pull/6837/commits/98621999699a16737ee73df59b5f17c880dfcb7c)

File: clickhouse-admin/api/src/lib.rs (hunk 1 above)
Comment (andrewjstone) [reply clickhouse-admin/api/src/lib.rs:90]: Good idea! Done in [9862199](https://github.com/oxidecomputer/omicron/pull/6837/commits/98621999699a16737ee73df59b5f17c880dfcb7c)

File: clickhouse-admin/api/src/lib.rs [hunk 6]
```diff
@@ -2,56 +2,38 @@
...
-
-#[derive(Debug, Serialize, Deserialize, JsonSchema)]
-pub struct KeeperConfigurableSettings {
-    /// A unique identifier for the configuration generation.
-    pub generation: Generation,
-    /// Configurable settings for a ClickHouse keeper node.
-    pub settings: KeeperSettings,
-}
 
+/// We separate the admin interface for the keeper and server APIs because they
```
Comment (sunshowers) [inline clickhouse-admin/api/src/lib.rs:14]: First paragraph of doc comments should be short (a lint for this is coming in Rust 1.83.) Could you just add something simple?

File: clickhouse-admin/api/src/lib.rs [hunk 7]
```diff
@@ -95,3 +77,28 @@ pub trait ClickhouseAdminApi {
         rqctx: RequestContext<Self::Context>,
     ) -> Result<HttpResponseOk<ClickhouseKeeperClusterMembership>, HttpError>;
 }
+
+/// We separate the admin interface for the keeper and server APIs because they
```
Comment (sunshowers) [inline clickhouse-admin/api/src/lib.rs:81]: Here as well

File: smf/clickhouse-admin-keeper/manifest.xml [hunk 8]
```diff
@@ -17,7 +17,7 @@
   </dependency>
 
   <exec_method type='method' name='start'
-    exec='ctrun -l child -o noorphan,regent /opt/oxide/clickhouse-admin/bin/clickhouse-admin run -c /var/svc/manifest/site/clickhouse-admin/config.toml -a %{config/http_address} -l %{config/ch_address} -b %{config/ch_binary}  &amp;'
+    exec='ctrun -l child -o noorphan,regent /opt/oxide/clickhouse-admin-keeper/bin/clickhouse-admin-keeper run -c /var/svc/manifest/site/clickhouse-admin-keeper/config.toml -a %{config/http_address} -l %{config/ch_address} -b %{config/ch_binary}  &amp;'
```
Comment (sunshowers) [inline smf/clickhouse-admin-keeper/manifest.xml:20]: Could I ask you to use the long version of command arguments here? ^_^

File: smf/clickhouse-admin-server/manifest.xml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  <service_fmri value='svc:/milestone/multi-user:default' />
+  </dependency>
+
+  <dependency name='zone_network_setup' grouping='require_all' restart_on='none'
+    type='service'>
+  <service_fmri value='svc:/oxide/zone-network-setup:default' />
+  </dependency>
+
+  <exec_method type='method' name='start'
+    exec='ctrun -l child -o noorphan,regent /opt/oxide/clickhouse-admin-server/bin/clickhouse-admin-server run -c /var/svc/manifest/site/clickhouse-admin-server/config.toml -a %{config/http_address} -l %{config/ch_address} -b %{config/ch_binary}  &amp;'
```
Comment (sunshowers) [inline smf/clickhouse-admin-server/manifest.xml:20]: And here.

**Review by sunshowers — APPROVED** [review]
Looks good! The new SMF definition looks very similar to the existing keeper one so I'll presume it's okay. Everything else looks pretty mechanical. Thanks!

File: clickhouse-admin/api/src/lib.rs (hunk 6 above)
Comment (andrewjstone) [reply clickhouse-admin/api/src/lib.rs:16]: Thanks for the tip! Done in [0a2e28b](https://github.com/oxidecomputer/omicron/pull/6837/commits/0a2e28b22169bf55587164395a620fb00bef3d4c)

File: clickhouse-admin/api/src/lib.rs (hunk 7 above)
Comment (andrewjstone) [reply clickhouse-admin/api/src/lib.rs:85]: Done in [0a2e28b](https://github.com/oxidecomputer/omicron/pull/6837/commits/0a2e28b22169bf55587164395a620fb00bef3d4c)

File: smf/clickhouse-admin-server/manifest.xml (hunk 9 above)
Comment (andrewjstone) [reply smf/clickhouse-admin-server/manifest.xml]: Done in [0a2e28b](https://github.com/oxidecomputer/omicron/pull/6837/commits/0a2e28b22169bf55587164395a620fb00bef3d4c)

File: smf/clickhouse-admin-keeper/manifest.xml (hunk 8 above)
Comment (andrewjstone) [reply smf/clickhouse-admin-keeper/manifest.xml]: Done in [0a2e28b](https://github.com/oxidecomputer/omicron/pull/6837/commits/0a2e28b22169bf55587164395a620fb00bef3d4c)

**Comment by andrewjstone** [comment]
Thanks for the great review @sunshowers!

**Pull request merged**